internal/manifests/targetallocator/configmap.go

```diff
@@ -113,7 +113,7 @@ func ConfigMap(params Params) (*corev1.ConfigMap, error) {
 		taConfig["prometheus_cr"] = prometheusCRConfig
 	}
 
-	if params.Config.CertManagerAvailability() == certmanager.Available && featuregate.EnableTargetAllocatorMTLS.IsEnabled() {
+	if params.Config.CertManagerAvailability == certmanager.Available && featuregate.EnableTargetAllocatorMTLS.IsEnabled() {
```

Comment:
Note this is a change in the behaviour: before, we were checking if cert-manager was available in each reconciliation. So, if it was installed/removed during two different reconciliations, the result would be different.

This applies to other changes introduced as part of this PR too.

I'm not against the change but we should, at least, add a changelog entry.

Reply:
I don't see the functional change. The difference here is we directly access the field, but the method just returns that field:
```
// CertManagerAvailability represents the availability of the Cert-Manager.
func (c *Config) CertManagerAvailability() certmanager.Availability {
	return c.certManagerAvailability
}
```

Reply:
Oh sorry. You are right. I thought this was calling the `autodetect` method (it was at some point).